src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs

```diff
@@ -38,10 +39,24 @@ public void OnBeginRequest(HttpContext httpContext, long timestamp)
         {
             if (this.client.IsEnabled())
             {
-                httpContext.Features.Set(new RequestTelemetry());
+                RequestTelemetry requestTelemetry = new RequestTelemetry();
+                if (string.IsNullOrEmpty(requestTelemetry.Context.InstrumentationKey))
+                {
+                    this.client.Initialize(requestTelemetry);
```

Comment:
At least in .net (full) sdk, this forces context initialization, which means requestTelemetry.Context.InstrumenationKey would be set if the client has the instrumentation key. Since requestTelemetry.Context.InstrumenationKey is used in the method below - I guess this call is here to make sure the value is present before using it.